### DISCUSSION TITLE
How to view container logs when app won't boot

### DISCUSSION THREAD

**Discussion by swelham** [basecamp/kamal#358] [Q&A] [answered]
Hi. I have been testing out mrsk using our production app and I can't get past the healthcheck stage of deployment. It looks like the container running our app is crashing on boot, however I cannot find any logs or terminal output to confirms that.

Is there a way to see what's going on during the healthcheck stage?

For reference these are the failing healthcheck logs.

```
INFO [61a0ac16] Running docker run --name traefik --detach --restart unless-stopped --publish 80:80 --volume /var/run/docker.sock:/var/run/docker.sock --log-opt max-size="10m" traefik:v2.9 --providers.docker --log.level="DEBUG" on [REDACTED]
  INFO [61a0ac16] Finished in 0.305 seconds with exit status 125 (failed).
Ensure app can pass healthcheck...
  INFO [97e8727e] Running docker run --detach --name healthcheck-my-app-4d09ccd4a9074a75f92d4c52cd071c384e4e45aa --publish 3999:4000 --label service=healthcheck-my-app -e MRSK_CONTAINER_NAME="healthcheck-my-app" -e --health-cmd "curl -f http://localhost:4000/api/v1/ping || exit 1" --health-interval "1s" registry.digitalocean.com/my-org-my-app-poc/my-org/my-app-poc:4d09ccd4a9074a75f92d4c52cd071c384e4e45aa on [REDACTED]
  INFO [a726fd7e] Running docker container ls --all --filter name=^healthcheck-my-app-4d09ccd4a9074a75f92d4c52cd071c384e4e45aa$ --quiet | xargs docker stop on [REDACTED]
  INFO [a726fd7e] Finished in 0.326 seconds with exit status 123 (failed).
  INFO [14fa59f2] Running docker container ls --all --filter name=^healthcheck-my-app-4d09ccd4a9074a75f92d4c52cd071c384e4e45aa$ --quiet | xargs docker container rm on [REDACTED]
  INFO [14fa59f2] Finished in 0.333 seconds with exit status 123 (failed).
  ```

**Comment by Jberczel** [comment, marked as answer]
Normally, app log and healthcheck log are outputted, but in your case, it looks like traefik failed to boot.

Unfortunately, if there's an issue with traefik, it doesn't raise an error:
https://github.com/mrsked/mrsk/blob/9a501867b438761239b32c9a8ae7c750aa4d12c6/lib/mrsk/cli/traefik.rb#L7

You may want to try and remote into your server and inspect the traefik container or check its logs to determine underlying issue.

**Reply by swelham** [reply]
Ah, I misunderstood that output. I thought traefik was up because I could see the container running on the server and logs had some entries about being in the `UP` status.

However, after digging back through the logs line by line I found an error about port config, which after fixing, it appears to be working now.

Thanks for the pointer 👍
